### ISSUE TITLE
A preinstall script should be written to signify which package-manager should be used

### ISSUE THREAD

**Issue by Cronus1007** [creativecommons/creativecommons-base#61]
## Description
Whenever we do a simple yarn install then bugs are reported while when we do a npm install nos uch bugs are found.
## Reproduction

1. git checkout docs
2. cd docs
3. yarn install
4. gyp error found 

## Screenshots

![Screenshot from 2021-01-10 20-17-34](https://user-images.githubusercontent.com/56436023/104599478-d77bc880-569d-11eb-9727-789b6282f74d.png)
## Environment

 - Device: laptop
 - OS: Ubuntu 20.02
 - Browser: Chrome

## Resolution

- [x] I would be interested in resolving this bug.

**Comment by Cronus1007** [comment]
@JackieBinya Plzzz have a look at the issue.

**Comment by Cronus1007** [comment]
@JackieBinya Here I am not setting up yarn. I am intend to add a `check_packagemanager.sh` preinstall script and while doing yarn install it will display an error to use npm rather than yarn.

**Comment by JackieBinya** [comment]
Thanks for the explanation @Cronus1007. I am unfamiliar with what I will assume is devops. Let me tag engineers who might be of better help to you.
 @hugosolar and @zackkrida please kindly assist @Cronus1007  resolve this issue, unfortunately it is beyond my scope of knowledge.

**Comment by Cronus1007** [comment]
@zackkrida Here what I intend to do is write a _scripts folder with `.sh` scripts to check the package manager. 
Yes the thing is quite obvious in this that we should run a `npm install` since it includes `package-lock.json` but when I first went through the codebase I missed it and I ran `yarn install` rather than `npm install` which yeilded `gyp-error`. So what this bash script will do is if anyone runs `yarn install` an error message will be displayed in the terminal.

**Comment by zackkrida** [comment]
HI @Cronus1007 this sounds good. In the future we could standardize this in all of our js projects using npm.

FYI, I've done this before with the following simple scripts. You don't have to do it this way, just an option:

```json
{
  "scripts": {
        "preinstall": "$npm_execpath --silent run checkyarn",
        "checkyarn": "echo \"$npm_execpath\" | grep -q \"npm\\.js$\" || (echo 'Please use npm instead of yarn.' && echo && exit 1)",
  }
}
```

**Comment by Cronus1007** [comment]
@zackkrida  Thanks. Yes I have an experience in writing the bash scripts so it won't be a problem for me .

**Comment by dhruvkb** [comment]
Personally I'd prefer the `checkyarn` script to be in `npm scripts` rather than a separate bash file.

**Comment by Cronus1007** [comment]
@dhruvkb  Okk I will follow the method as prescribed by @zackkrida

**Comment by Pavankalyan0105** [comment]
Hi i would like to work on this issue. can anyone please help me to how to get started?

**Comment by Cronus1007** [comment]
@Pavankalyan0105  The issue is on the verge of being closed and I have already atttached the PR #62  for solving the issue. But you can have a look at the other issues.

**Comment by zackkrida** [comment]
Fixed via #62
